streams/src/test/java/org/apache/kafka/streams/state/internals/TimestampedKeyValueStoreBuilderTest.java

```diff
@@ -154,17 +154,34 @@ public void shouldThrowNullPointerIfInnerIsNull() {
     }
 
     @Test
-    public void shouldThrowNullPointerIfKeySerdeIsNull() {
-        assertThrows(NullPointerException.class, () -> new TimestampedKeyValueStoreBuilder<>(supplier, null, Serdes.String(), new MockTime()));
+    public void shouldNotThrowNullPointerIfKeySerdeIsNull() {
```

Comment:
These test changes are not directly related to this PR. While I was in here looking for inspiration for the new VersionedKeyValueStoreBuilderTest.java tests, I noticed that these tests were broken. The only reason they were throwing NullPointerExceptions is because the supplier was not being mocked to return name and metrics scope. Once added, then passing null key or value serdes no longer throws an exception.

I also added the same required mocking into `shouldThrowNullPointerIfTimeIsNull()` below so that the test actually throws an NPE on time being null, and not supplier return null for name or metrics scope.